bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,1752 @@
+        self.impurities_content = impurity_content.values()
+
+        self.imp_data_t_ref = [data["T_ref"] for key, data in impurity_data.items()]
+        self.imp_data_l_ref = [data["L_ref"] for key, data in impurity_data.items()]
+
+        # Adimensional radius at the mid-plane.
+        # From the core to the last closed flux surface
+        self.rho_core = self.collect_rho_core_values()
+
+        # For each flux tube, density and temperature at the mid-plane
+        self.ne_mp, self.te_mp = self.core_electron_density_temperature_profile(
+            self.rho_core
+        )
+
+    def build_mp_rad_profile(self):
+        """
+        1D profile of the line radiation loss at the mid-plane
+        through the scrape-off layer
+        """
+        # Radiative loss function values for each impurity species
+        loss_f = [
+            self.radiative_loss_function_values(self.te_mp, t_ref, l_ref)
+            for t_ref, l_ref in zip(self.imp_data_t_ref, self.imp_data_l_ref)
+        ]
+
+        # Line radiation loss. Mid-plane distribution through the SoL
+        rad = [
+            self.calculate_line_radiation_loss(self.ne_mp, loss, fi)
+            for loss, fi in zip(loss_f, self.impurities_content)
+        ]
+
+        return self.plot_1d_profile(self.rho_core, rad)
+
+    def build_core_radiation_map(self):
+        """
+        2D map of the line radiation loss in the plasma core.
+        """
+        # Closed flux tubes within the separatrix
+        flux_tubes = self.collect_flux_tubes(self.rho_core)
+
+        # For each flux tube, poloidal density profile.
+        ne_pol = [
+            self.flux_tube_pol_n(ft, n, core=True)
+            for ft, n in zip(flux_tubes, self.ne_mp)
+        ]
+
+        # For each flux tube, poloidal temperature profile.
+        te_pol = [
+            self.flux_tube_pol_t(ft, t, core=True)
+            for ft, t in zip(flux_tubes, self.te_mp)
+        ]
+
+        # For each impurity species and for each flux tube,
+        # poloidal distribution of the radiative power loss function.
+        loss_f = [
+            [self.radiative_loss_function_values(t, t_ref, l_ref) for t in te_pol]
+            for t_ref, l_ref in zip(self.imp_data_t_ref, self.imp_data_l_ref)
+        ]
+
+        # For each impurity species and for each flux tube,
+        # poloidal distribution of the line radiation loss.
+        rad = [
+            [
+                self.calculate_line_radiation_loss(n, l_f, fi)
+                for n, l_f in zip(ne_pol, ft)
+            ]
+            for ft, fi in zip(loss_f, self.impurities_content)
+        ]
+
+        # Total line radiation loss along each flux tube
+        total_rad = np.sum(np.array(rad, dtype=object), axis=0).tolist()
+
+        return self.plot_2d_map(flux_tubes, total_rad)
+
+
+class STScrapeOffLayer(ScrapeOffLayerSector, ScrapeOffLayer, TwoPointModelTools):
+    """
+    Specific class for the scrape-off layer emission of Spherical Tokamak
```

Comment:
Ditto here?

Reply:
Similarly. I collected everything in just one class, and left just what is strictly necessary for a DN in another class.